server/src/main/java/org/apache/iotdb/db/engine/compaction/reader/PointPriorityReader.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.engine.compaction.reader;
+
+import org.apache.iotdb.commons.exception.IllegalPathException;
+import org.apache.iotdb.db.engine.compaction.cross.rewrite.task.FastCompactionPerformerSubTask;
+import org.apache.iotdb.db.engine.compaction.cross.utils.PageElement;
+import org.apache.iotdb.db.engine.compaction.cross.utils.PointElement;
+import org.apache.iotdb.db.exception.WriteProcessException;
+import org.apache.iotdb.tsfile.read.common.IBatchDataIterator;
+import org.apache.iotdb.tsfile.utils.Pair;
+import org.apache.iotdb.tsfile.utils.TsPrimitiveType;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.PriorityQueue;
+
+/**
+ * This reader is used to deduplicate and organize overlapping pages, and read out points in order.
+ * It is used for compaction.
+ */
+public class PointPriorityReader {
+  private long lastTime;
+
+  private final PriorityQueue<PointElement> pointQueue;
+
+  private final FastCompactionPerformerSubTask.RemovePage removePage;
+
+  private Pair<Long, Object> currentPoint;
+
+  private boolean isNewPoint = true;
+
+  private List<PageElement> newOverlappedPages;
```

Comment:
The usage and update of this variable is outside this class, it should be moved outside.

Reply:
Resolved. It has been renamed to `candidateOverlappedPages` and put into `FastCompactionPerformerSubTask` class as a global variable. Pages in this list will be sequentially judged whether there is a real overlap to choose whether to put them in the point priority reader to deserialize or directly flush to chunk writer. During the process of compacting overlapped page, there may be new overlapped pages added into this list.